File: Helló, Liskov!/Liskovsertes.java
class tank
{
	 public void javit()
	 {
		System.out.print("Tudok Lánctalpat Javítani!\n");
	 }
}

class ebr extends tank // Az ebr tud lánctalpat javítani
{
}

public class Liskovsert
{
	 public static void fgv(tank tank)
	 {
		  tank.javit();
	 }

	 public static void main(String[] args)
	{
        tank ebr = new ebr();
        
        fgv(ebr);
		 // sérül az LSP, mert a P::fgv megjavítaná az ebr lánctalpát, ami lehetetlen mert az ebr kerekes

	}
}

File: Helló, Liskov!/Liskovsertes.cpp
#include <iostream>

using namespace std;
// ez a T az LSP-ben
class tank {
public:
     virtual void javit() {
	cout<<"Tudok Lánctalpat Javítani!"<<endl;
     };
};


class Program {
public:
     void fgv ( tank &tank ) {
          tank.javit();
     }
};



class Pingvin : public tank //  Az ebr tud lánctalpat javítani
{};

int main ( int argc, char **argv )
{
     Program program;
     tank tank;
     program.fgv ( tank );

     Pingvin pingvin;
     program.fgv ( pingvin ); // sérül az LSP, mert a P::fgv megjavítaná az ebr lánctalpát, ami lehetetlen mert az ebr kerekes

}

